src/role-access/services/allocate-role.service.ts

```diff
@@ -24,6 +26,18 @@ export class AllocateRoleService {
     }
   }
 
+  public specificAccessApproval(specificAccessStateData: SpecificAccessStateData, dtperiod: Period): Observable<any> {
```

Comment:
just a suggestion - there are lot of "any" references in the PR and code base, it might take a lot of time to refactor this now, it is best we create a dev debt ticket to address this issue.